tests/tools/saunafsXX.sh

```diff
@@ -4,117 +4,135 @@
 # TODO: Don't use build information in debian package versions. It causes
 # problems when downgrading, because you need to include the build information
 # in the downgrade command.
-SAUNAFSXX_TAG_APT="4.0.1-20240301-164017-stable-main-82161d4f"
-SAUNAFSXX_TAG="4.0.1"
+
+SAUNAFSXX_TAG="4.1.0"
 
 install_saunafsXX() {
-	rm -rf "$SAUNAFSXX_DIR"
-	mkdir -p "$SAUNAFSXX_DIR"
+	rm -rf "${SAUNAFSXX_DIR}"
+	mkdir -p "${SAUNAFSXX_DIR}"
 	local distro="$(lsb_release -si)"
-	case "$distro" in
+	case "${distro}" in
 	Ubuntu | Debian)
-		local codename="$(lsb_release -sc)"
-		mkdir -p ${TEMP_DIR}/apt/apt.conf.d
-		mkdir -p ${TEMP_DIR}/apt/var/lib/apt/partial
-		mkdir -p ${TEMP_DIR}/apt/var/cache/apt/archives/partial
-		mkdir -p ${TEMP_DIR}/apt/var/lib/dpkg
-		cp /var/lib/dpkg/status ${TEMP_DIR}/apt/var/lib/dpkg/status
-		cat >${TEMP_DIR}/apt/apt.conf <<END
+		local distro_id="$(lsb_release -si | tr '[:upper:]' '[:lower:]' | tail -1)"
+		local codename="$(lsb_release -sc | tail -1)"
+		local release="$(lsb_release -sr | tail -1)"
+		mkdir -p "${TEMP_DIR}/apt/apt.conf.d"
+		mkdir -p "${TEMP_DIR}/apt/var/lib/apt/partial"
+		mkdir -p "${TEMP_DIR}/apt/var/cache/apt/archives/partial"
+		mkdir -p "${TEMP_DIR}/apt/var/lib/dpkg"
+		mkdir -p "${TEMP_DIR}/usr/share/keyrings"
+		cp /var/lib/dpkg/status "${TEMP_DIR}/apt/var/lib/dpkg/status"
+		cat >"${TEMP_DIR}/apt/apt.conf" <<END
 Dir::State "${TEMP_DIR}/apt/var/lib/apt";
 Dir::State::status "${TEMP_DIR}/apt/var/lib/dpkg/status";
 Dir::Etc::SourceList "${TEMP_DIR}/apt/saunafs.list";
 Dir::Cache "${TEMP_DIR}/apt/var/cache/apt";
 Dir::Etc::Parts "${TEMP_DIR}/apt/apt.conf.d";
 END
 		local destdir="${TEMP_DIR}/apt/var/cache/apt/archives"
+		echo "deb [arch=amd64 signed-by=/usr/share/keyrings/saunafs-archive-keyring.gpg] https://repo.saunafs.com/repository/saunafs-${distro_id}-${release}/ ${codename} main" >"${TEMP_DIR}/apt/saunafs.list"
```

Comment:
I don't think this is really needed: The apt repository is already setup by setup_machine.sh, and this would conflict if the repository was setup in any other way (for example, different place of keyring or different architecture):

https://github.com/leil-io/saunafs/blob/a7cb56694148bb46a3788949e1f4f2d385ecd1b1/tests/setup_machine.sh#L317